File: mage/src/main/java/mil/nga/giat/mage/filter/Filter.java
package mil.nga.giat.mage.filter;

import com.j256.ormlite.stmt.QueryBuilder;
import com.j256.ormlite.stmt.Where;

import java.sql.SQLException;

public interface Filter<T> {

    public QueryBuilder<?, Long> query() throws SQLException;
    public void and(Where<? extends T, Long> where) throws SQLException;

    public boolean passesFilter(T obj);
    
}

File: mage/src/main/java/mil/nga/giat/mage/filter/Filter.kt
package mil.nga.giat.mage.filter

import com.j256.ormlite.stmt.QueryBuilder
import com.j256.ormlite.stmt.Where
import java.sql.SQLException

interface Filter<in T> {
   @Throws(SQLException::class)
   fun query(): QueryBuilder<*, Long>?

   @Throws(SQLException::class)
   fun and(where: Where<*, Long>)

   fun passesFilter(obj: T): Boolean
}